js/evidence.js

```diff
@@ -1,26 +1,196 @@
+				process_column_backgrounds();
+				$story_container.data('page', data.page + 1);
+			} else if ('filtered' === type) {
+				$story_container.html($.parseJSON(response));
+				process_column_backgrounds();
+				$story_container.data('page', 1);
+			}
+			$('.new').show(400);
+		}).always(function () {
+			fetching = false;
+		});
+	}
+
+	/**
+	 * Load more posts when the footer is reached.
+	 */
+	function infinite_scroll() {
+		window.addEventListener('scroll', function () {
+			if (fetching ||
+					all_stories ||
+					!footer_is_visible() ||
+					$story_container.data('page') === $story_container.data('total-pages') ||
+					'' !== $('#filter-options').find('option:selected').val()) {
+				return;
+			}
+
+			var data = {
+					action: 'evidence_stories',
+					nonce: evidence.nonce,
+					page: $story_container.data('page')
+				};
+
+			fetch_and_display_stories(data, 'scrolled');
+		}, 500);
+	}
+
+	/**
+	 * Scroll to the filter wrapper when an option is selected.
+	 */
+	function scroll_to_filter() {
+		var filters_top = Math.round($filters.offset().top - offset);
+
+		if ($(window).scrollTop() !== filters_top) {
+			$('html, body').animate({
+				scrollTop: filters_top
+			}, 750);
+		}
+	}
+
+	/**
+	 * Update the header text and document title when an option is selected from the filter drop-down.
+	 */
+	function update_title(name, value) {
+		var heading = value ? name : ' ',
+			doc_title = value ? name + ' | ' + evidence.default_title : evidence.default_title;
+
+		$('.topic-title').find('h3').html(heading);
+
+		document.title = doc_title;
+	}
 
-	$(document).ready( function() {
+	/**
+	 * Update the history state.
+	 */
+	function update_history(name, value) {
+		var state = { name: name, value: value },
+			url = value ? evidence.default_url + 'category/' + value + '/' : evidence.default_url;
+
+		history.pushState(state, null, url);
+	}
+
+	$(document).ready(function () {
+		var option = $(this).find('option:selected'),
+			value = option.val(),
+			name = value ? option.text() : '&nbsp;',
+			state = { name: name, value: value };
+
+		history.replaceState(state, null, null);
 		process_column_backgrounds();
+		fix_filters();
+		infinite_scroll();
 	});
 
-}(jQuery, window));
+	$options.on('change', function () {
+		var option = $(this).find('option:selected'),
+			name = option.text(),
+			value = option.val(),
+			data = {
+				action: 'evidence_stories',
+				nonce: evidence.nonce,
+				category: value
+			};
+
+		scroll_to_filter();
+		update_title(name, value);
+		update_history(name, value);
+		fetch_and_display_stories(data, 'filtered');
+	});
+
+	window.onpopstate = function(event) {
+		var name = event.state.name,
+			value = event.state.value,
+			data = {
+				action: 'evidence_stories',
+				nonce: evidence.nonce,
+				category: value
+			};
+
+		$options.val(value);
+		update_title(name, value);
+		fetch_and_display_stories(data, 'filtered');
+	};
+
+}(jQuery, window, document, evidence));
```

Comment:
I get `evidence is not defined` in the style guide because `evidence` is not defined. :) Not really a big deal as replacing all of the ajax functionality that WP provides would be kind of ridiculous. The file could start with `var evidence = evidence || {};` as the first line so that it is available no matter what. There may still be other problems, but it would help avoid the first error.